playbooks/common/openshift-master/config.yml

```diff
@@ -337,12 +361,54 @@
     with_items: openshift_master_named_certificates
     when: named_certs_specified | bool
 
-- name: Configure master instances
-  hosts: oo_masters_to_config
+- name: Configure first master instance
```

Comment:
> there will still be a need for the openshift_docker role, but splitting out the OpenShift version handling, etc into a separate role would make a lot of sense.

Can you clarify what the need is for openshift_docker? I think there's literally nothing left in my branch beyond setting this version. Do we need it as a wrapper around docker for some reason,  or should I just flip all the callers to just require "docker" and rename it to openshift_version.

> I think my hangup here is what determines we are upgrading? This becomes even murkier when you consider the idealized stand-alone Ansible role use case:

Per our discussions the playbook (i.e. orchestrator) has full control over this. Roles will do what they're told, or if they're not told any version, they'll install latest. For config playbooks it behaves as described above, but the major upgrade playbook will specify openshift_version=3.3 for first master, run openshift_docker role to convert that to 3.3.y.z, and then re-use that value for all the remaining hosts. A minor upgrade playbook can do similar.

Does that help at all? 

If not I can ditch the "don't mess with running versions unless upgrading" goal we planned, and instead focus on keeping the cluster in sync with first master. (possibly with some version comparison and related actions, I'll have to think on that a bit more)